flow/surface_frame.h

```diff
@@ -21,17 +21,52 @@ class SurfaceFrame {
   using SubmitCallback =
       std::function<bool(const SurfaceFrame& surface_frame, SkCanvas* canvas)>;
 
+  // Information about the underlying framebuffer
+  struct FramebufferInfo {
+    // Indicates whether or not the surface supports pixel readback as used in
+    // circumstances such as a BackdropFilter.
+    bool supports_readback = false;
+
+    // This is the area of framebuffer that lags behind the front buffer.
+    //
+    // Correctly providing exiting_damage is necessary for supporting double and
+    // triple buffering. Embedder is responsible for tracking this area for each
+    // of the back buffers used. When doing partial redraw, this area will be
+    // repainted alongside of dirty area determined by diffing current and
+    // last successfully rasterized layer tree;
+    //
+    // If existing damage is unspecified (empty vector), entire frame will be
+    // rasterized (no partial redraw). To signal that there is no existing
+    // damage use vector with an empty SkIRect.
+    std::vector<SkIRect> existing_damage;
+  };
+
   SurfaceFrame(sk_sp<SkSurface> surface,
-               bool supports_readback,
+               FramebufferInfo framebuffer_info,
                const SubmitCallback& submit_callback);
 
   SurfaceFrame(sk_sp<SkSurface> surface,
-               bool supports_readback,
+               FramebufferInfo framebuffer_info,
                const SubmitCallback& submit_callback,
                std::unique_ptr<GLContextResult> context_result);
 
   ~SurfaceFrame();
 
+  struct SubmitInfo {
```

Comment:
Given that this will be a vector of max-size 1 currently, what do you think about making these fields an `std::optional<SkIRect>` instead. When we decide to make the optimization of having multiple damage rects we can make these into vectors.

Reply:
I'm fine with this being either a list, or a single rect, but I think `SubmitInfo` should be consistent with `FramebufferInfo`, either both should contain a region or a single rect.

Reply:
Agreed, lets make both of them a single rect for now.